File: src/Guias/TP3/Ej9/Point.java
package Guias.TP3.Ej9;

public class Point {
    private double x,y;
    Point(double x,double y){
        this.x=x;
        this.y=y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }
    public double distance(Point b){
        return Math.sqrt(Math.pow(x-b.getX(),2)+Math.pow(y- b.getY(),2));
    }
    public double distanceX(Point b){
        return Math.abs(x-b.getX());
    }
    public double distanceY(Point b){
        return Math.abs(y- b.getY());
    }
}

File: Ruby/Guias/G7/Ej4/point.rb
class Point
  def initialize(x, y)
    @x = x
    @y = y
  end
  def distance(point)
    ((point.x - @x)**2 + (point.y - @y)**2)**0.5
  end
  def distance_x(point)
    (self.x-point.x).abs
  end
  def distance_y(point)
    (self.y-point.y).abs
  end
  def to_s
    "{#{@x}, #{@y}}"
  end
  protected
  attr_reader :x, :y
end
